### DISCUSSION TITLE
option to schedule

### DISCUSSION THREAD

**Discussion by richenberg** [mtkennerly/ludusavi#197] [Ideas]
possible to make a option to schedule it, i want it to run one time per week

**Comment by mtkennerly** [comment]
This is possible! Since Ludusavi has a command line interface (CLI), you just need something that runs the CLI every so often.

If you're using Windows, you can create a service. Someone wrote an example of how to do this with NSSM here: https://github.com/mtkennerly/ludusavi/issues/90#issuecomment-1454204019. You could also do something similar with [Shawl](https://github.com/mtkennerly/shawl) (one of my other projects), although it doesn't have a GUI like NSSM does.

If you're using Linux, you could do this with cron or systemd. Someone wrote an example of how to do this on the Steam Deck with systemd: https://www.reddit.com/r/SteamDeck/comments/108o2yt/backup_saves_automatic_with_ludusavi_daemon/

**Comment by chennin** [comment]
**Linux Unprivileged Systemd method**

Recommend setting up Ludusavi via the GUI first.

* Create a service with `systemctl edit --user --force --full ludusavi-backup.service`

```ini
[Unit]
Description=Back up game saves
After=network.target

[Service]
Type=oneshot
ExecStart=/usr/bin/ludusavi backup --force

[Install]
WantedBy=default.target
```

* Create a timer with `systemctl edit --user --force --full ludusavi-backup.timer` . The following runs daily, or 5-10 minutes after boot.

```ini
[Unit]
Description=Back up game saves
RefuseManualStart=no
RefuseManualStop=no

[Timer]
OnCalendar=daily
#Persistent=true
RandomizedDelaySec=5min
OnBootSec=5min

[Install]
WantedBy=timers.target
```

* Enable the timer and run a backup: `systemctl --user enable --now ludusavi-backup.timer`

List timers: `systemctl --user list-timers --all`

Run a backup manually: `systemctl --user start ludusavi-backup.service`

Check logs: `journalctl --user -u ludusavi-backup.service`

**Comment by CrawliestCotter** [comment]
For what it's worth, here's a systemd service file to run a backup every 5 minutes for the flatpak version of ludusavi:

#### Run:
`systemctl edit --user  --force --full ludusavi-flatpak.service`


#### Then copy the following into your editor:
```
[Unit]
Description=Ludusavi Flatpak Backup

[Service]
Type=idle
Restart=always
RestartSec=300s
ExecStart=flatpak run com.github.mtkennerly.ludusavi backup --force

[Install]
WantedBy=default.target
```
#### Finally run:
`systemctl --user enable ludusavi-flatpak.service` to enable the service.

I created github repo for the service here:
https://github.com/CrawliestCotter/Ludusavi-Flatpak-Systemd-Service
